spec.bs

```diff
@@ -5677,6 +5873,9 @@ An <dfn>interest group ad</dfn> is a [=struct=] with the following [=struct/item
     with registered macros. Each origin's [=origin/scheme=] must be "`https`" and each origin must be
     <a href="https://github.com/privacysandbox/attestation">enrolled</a>. Only meaningful in
     [=interest group/ads=], but ignored in [=interest group/ad components=].
+  : <dfn> ad render ID</dfn>

```

Comment:
seems need to update https://github.com/WICG/turtledove/blob/main/spec.bs#L1419, since ad render id is in prevWin's ad json?

Reply:
Yes, though perhaps that can be in another CL. On a related note, is there a reason why we are specifying that we are using JSON to save previous wins? Perhaps just removing the note that we store it as JSON would be sufficient.

Reply:
can you post a link to the place you're refering to? There are [IDL](https://wicg.github.io/turtledove/#dictdef-previouswin) and [struct](https://wicg.github.io/turtledove/#previous-win) for previous win, whose ad json fields are json, not previous wins are json, right?

Reply:
Yes, the "ad JSON" field should probably be an "interest group ad" instead, since we need access to the fields.

Reply:
I think it was following https://crsrc.org/c/content/services/auction_worklet/public/mojom/bidder_worklet.mojom;drc=82dff63dbf9db05e9274e11d9128af7b9f51ceaa;l=37. The field in browser signals should be json as well?

Reply:
Yes, internally we store it as JSON, but we convert it to a JavaScript Object before providing it in generateBid (see https://crsrc.org/c/content/services/auction_worklet/bidder_lazy_filler.cc;l=68 ).

So it being stored as JSON doesn't really affect the public interface -- it seems more like an implementation detail that doesn't belong in the spec.

Reply:
Yeah the spec is wrong then, since[ it passes](https://github.com/WICG/turtledove/blob/main/spec.bs#L3904) a DOMString to generateBid for this field in browser signals . We need to change the previous win struct's [ad json](https://wicg.github.io/turtledove/#previous-win-ad-json) field to a struct, and PreviousWin IDL's [adJson](https://wicg.github.io/turtledove/#dom-previouswin-adjson) to a dictionary of several fields.

Reply:
this can be a separate PR and I can do that if you'd like. And then I'm good with landing this one.